New Test set & other minor improvements

ajn opened the pull request:
- Added `ginkgo/gomega` based test set
- Added `URLString` for normalizing URIs
- Updated original inline linear example to include `Extensions`

dim committed:
Starting new test suite

dim committed:
First test

ajn committed:
Added test for vast_inline_nonlinear.xml

ajn committed:
Added URIString to help normalize URIs

ajn committed:
Implemented URIString, cleaning up parsed URIs

ajn committed:
Added `wrapper linear` test

ajn committed:
Added test for Extensions

dim commented on uri_string.go:
```diff
@@ -0,0 +1,17 @@
+
+// UnmarshalText implements the encoding.TextUnmarshaler interface.
+func (s *URIString) UnmarshalText(data []byte) error {
+	*s = URIString(strings.TrimSpace(string(data)))
```
https://golang.org/pkg/bytes/#TrimSpace

dim commented on uri_string.go:
```diff
@@ -0,0 +1,17 @@
+import "strings"
+
+// URIString is a string that allows for stripping of whitespace when Unmarshalled
+type URIString string
```
Any reason not to call it just `URI`?

dim commented on vast_test.go:
```diff
@@ -2,310 +2,291 @@ package vast
+					},
+					Description: "VAST 2.0 Instream Test 1",
+					Error: []Error{
+						{URI: URIString("http://myErrorURL/error")},
```
you don't need to cast, go knows what type it is

ajn committed:
Review changes

- Renamed `URIString` to `URI`
- Using `bytes.TrimSpace` instead of `strings.TrimSpace` in `URI.Unmarshal`
- Removing the unnecessary wrapping of URIs in tests
